dlt/pipeline/pipeline.py

```diff
@@ -474,11 +474,13 @@ def _verify_destination_capabilities(
 
         # verify merge strategy
         for table in self.default_schema.data_tables(include_incomplete=True):
-            # temp solution to prevent raising exceptions for destinations such as
-            # `fileystem` and `weaviate`, which do handle the `merge` write
-            # disposition, but don't implement any of the defined merge strategies
-            if caps.supported_merge_strategies is not None:
-                if "x-merge-strategy" in table and table["x-merge-strategy"] not in caps.supported_merge_strategies:  # type: ignore[typeddict-item]
+            if "x-merge-strategy" in table and table["x-merge-strategy"] not in caps.supported_merge_strategies:  # type: ignore[typeddict-item]
+                if self.destination.destination_name in ("filesystem", "weaviate") and table["x-merge-strategy"] == "delete-insert":  # type: ignore[typeddict-item]
```

Comment:
Current behaviour of `merge` with `filesystem` destination:
1. `write_disposition="merge"` (which implicitly defaults to `delete-insert` merge strategy) ➜ fall back to `append`
2. `write_disposition={"disposition="merge", strategy="upsert"}` ➜ `upsert` merge strategy

I'm not a fan of this.

I think (1) should raise an error, because `delete-insert` is not implemented for `filesystem`.